File: objc/Facade/Facade/DataSpliter.m
#import "DataSpliter.h"

@implementation DataSpliter {
    int packetSize;
}

-(instancetype)initWithPacketSize:(int)size {
    if(self = [super init]) {
        packetSize = size;
    }
    return self;
}

-(NSArray*)splitData:(NSString*)data {
    int dataLen = (int)data.length;
    NSMutableArray* retList = [[NSMutableArray alloc] init];
    for (int i=0; i < dataLen; i += packetSize) {
        int len = (i + packetSize > dataLen) ? (dataLen - i) : packetSize;
        [retList addObject:[data substringWithRange:NSMakeRange(i, len)]];
    }
    return retList;
}

@end

File: swift/Facade/Facade/DataSpliter.swift
import Foundation

class DataSpliter {
    let packetSize: Int
    
    init(_ packetSize: Int) {
        self.packetSize = packetSize
    }
    
    func splitData(_ data: String) -> [String] {
        var retList = [String]()
        let sequence = stride(from: 0, to: data.count, by: packetSize)
        for i in sequence {
            let start = data.index(data.startIndex, offsetBy: i)
            let end = i + packetSize >= data.count ?
                data.index(data.endIndex, offsetBy: 0) :
                data.index(data.startIndex, offsetBy: i+packetSize)
            retList.append(String(data[start..<end]))
        }
        return retList
    }
}
